cmd/sync/sync.go

```diff
@@ -23,12 +27,40 @@ import (
 
 type syncFlags struct {
 	// project files polling interval
-	interval time.Duration
-	full     bool
-	watch    bool
-	output   flags.Output
-	exclude  []string
-	include  []string
+	interval    time.Duration
+	full        bool
+	watch       bool
+	output      flags.Output
+	exclude     []string
+	include     []string
+	excludeFrom string
+}
+
+func (f *syncFlags) readExcludeFrom(ctx context.Context) ([]string, error) {
+	if f.excludeFrom == "" {
+		return nil, nil
+	}
+
+	data, err := os.ReadFile(f.excludeFrom)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read exclude-from file: %w", err)
+	}
+
+	var patterns []string
+	scanner := bufio.NewScanner(bytes.NewReader(data))
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		// Skip empty lines and comments
```

Comment:
This seems to duplicate functionality already present in go-gitignore

https://github.com/sabhiram/go-gitignore/blob/master/ignore.go#L182

```
// CompileIgnoreFile uses an ignore file as the input, parses the lines out of
// the file and invokes the CompileIgnoreLines method.
func CompileIgnoreFile(fpath string) (*GitIgnore, error) {
```

Pattern function strips comments and empty lines:
https://github.com/sabhiram/go-gitignore/blob/master/ignore.go#L77

```
	// Strip comments [Rule 2]
	if strings.HasPrefix(line, `#`) {
		return nil, false
	}

	// Trim string [Rule 3]
	// TODO: Handle [Rule 3], when the " " is escaped with a \
	line = strings.Trim(line, " ")
```

Reply:
nice catch! removed this check in the follow-up commit